File: TDDMicroExercises.ProposedSolution/Java/TirePressureMonitoringSystem/src/test/java/tddmicroexercises/tirepressuremonitoringsystem/StubAlarm.java
package tddmicroexercises.tirepressuremonitoringsystem;


public class StubAlarm implements IAlarm
{
    private boolean alarmOnReturnValue;

    public void setAlarmOn(boolean returnValue)
    {
        alarmOnReturnValue = returnValue;
    }

    public void check()
    {
        return;
    }

    public boolean isAlarmOn()
    {
        return alarmOnReturnValue;
    }
}

File: CSharp/TirePressureMonitoringSystem.Tests/StubAlarm.cs
namespace TDDMicroExercises.TirePressureMonitoringSystem.Tests
{
    public class StubAlarm : IAlarm
    {
        private bool _alarmOnReturnValue;

        public void StubGetAlarmOn(bool returnValue)
        {
            _alarmOnReturnValue = returnValue;
        }

        void IAlarm.Check()
        {
            return;
        }

        bool IAlarm.AlarmOn
        {
            get { return _alarmOnReturnValue; }
        }
    }
}
